ts/components/AddServerDialog.tsx

```diff
@@ -0,0 +1,273 @@
+import React from 'react';
+
+import { SessionModal } from './session/SessionModal';
+import { SessionButton } from './session/SessionButton';
+import { SessionSpinner } from './session/SessionSpinner';
+
+interface Props {
+  i18n: any;
+  onClose: any;
+}
+
+interface State {
+  title: string;
+  error: string | null;
+  connecting: boolean;
+  success: boolean;
+  view: 'connecting' | 'default';
+  serverUrl: string;
+}
+
+export class AddServerDialog extends React.Component<Props, State> {
+  constructor(props: any) {
+    super(props);
+
+    this.state = {
+      title: this.props.i18n('addServerDialogTitle'),
+      error: null,
+      connecting: false,
+      success: false,
+      view: 'default',
+      serverUrl: '',
+    };
+
+    this.showError = this.showError.bind(this);
+    this.showView = this.showView.bind(this);
+    this.attemptConnection = this.attemptConnection.bind(this);
+
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+  }
+
+  public render() {
+    const { i18n } = this.props;
+
+    return (
+      <SessionModal
+        title={this.state.title}
+        onOk={() => null}
+        onClose={this.closeDialog}
+      >
+        {this.state.view === 'default' ? (
```

Comment:
tslint won't accept this

Reply:
also, when you don't have a `false` case as here: `: null`, you can just use && rather than the terniary operaotr